Glottis.cpp

```diff
@@ -291,26 +310,56 @@ bool Glottis::readFromXml(XmlNode &rootNode)
     for (i=0; i < (int)controlParamNode->childElement.size(); i++)
     {
       node = controlParamNode->childElement[i];
-      index = node->getAttributeInt("index");
-      name  = node->getAttributeString("name");
-      value = node->getAttributeDouble("value");
-
-      if ((index < 0) || (index >= (int)controlParam.size()))
+      try
       {
-        printf("Error: Control parameter index out of range for parameter '%s'!\n", name.c_str());
-        return false;
+          //index = node->getAttributeInt("index");
+          name = node->getAttributeString("name");
+          //value = node->getAttributeDouble("value"); //value does not exist anymore in glottis definition, set to neutral instead
+          min = node->getAttributeDouble("min");
+          max = node->getAttributeDouble("max");
+          neutral = node->getAttributeDouble("neutral");
       }
-
-      if (name != controlParam[index].name)
+      catch (const std::exception & e)
       {
-        printf("Error: The name of the control parameter %d is '%s' but should be '%s'!\n", 
-          index, name.c_str(), controlParam[index].name.c_str());
-        return false;
+          printf("Error: %s\n", e.what());
+          return false;
+      }
+      n = -1;
+      for (m = 0; (m < controlParam.size()) && (n == -1); m++)
+      {
+          if (name == controlParam[m].name)
+          {
+              n = m;
+          }
+      }
+      if (n != -1)
+      {
+          controlParam[n].min = min;
+          controlParam[n].max = max;
+          controlParam[n].neutral = neutral;
+          controlParam[n].x = neutral;
+          // Don't overwrite with the values from the xml-file. Keep the
```

Comment:
This should be discussed with Peter if it is intentional behavior for some reason. If not, read from XML.

Reply:
Discussed with Peter, the commented line can be removed. (I removed them in the new_speakerfile_branch).